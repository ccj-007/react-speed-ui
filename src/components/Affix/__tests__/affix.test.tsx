import { render } from '@testing-library/react';
import { Affix, Button } from '../../index';
import "@testing-library/jest-dom/extend-expect";
import React from 'react';

describe('test affix component', () => {
  it('should render a <Affix/> components', () => {
    const { asFragment } = render(
      <>
        <Affix offsetTop={20}>
          <Button>Affix top</Button>
        </Affix>
        <Affix offsetBottom={40}>
          <Button>Affix bottom</Button>
        </Affix>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  })

  it('should use speed-ui classNames', () => {
    const { container } = render(
      <Affix offsetTop={20}>
        <Button>Affix top</Button>
      </Affix>
    );
    expect(container.firstChild).toHaveClass('speed-affix speed-affix-top')
  })
});
