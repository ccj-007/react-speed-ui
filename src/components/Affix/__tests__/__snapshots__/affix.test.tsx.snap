// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test affix component should render a <Affix/> components 1`] = `
<DocumentFragment>
  <div
    class="speed-affix speed-affix-top"
    style="top: 20px; bottom: 0px;"
  >
    <div
      offsettop="20"
    >
      <button
        class="speed-button speed-button-primary"
        data-testid="button"
      >
        Affix top
      </button>
    </div>
  </div>
  <div
    class="speed-affix speed-affix-bottom"
    style="top: 0px; bottom: 40px;"
  >
    <div
      offsetbottom="40"
    >
      <button
        class="speed-button speed-button-primary"
        data-testid="button"
      >
        Affix bottom
      </button>
    </div>
  </div>
</DocumentFragment>
`;
